### PR TITLE
Addition of user-agent to request header - LXCA 4973

### PR DESCRIPTION
**Pull request by ManasaRaoK**
The ruby client would not be responsible for defining the value of the header.  
•	the client provide an optional parameter named user_agent_label to the connection method
•	By default , the library will use  'ruby  <gem version number>'
•	The library will append  ‘ (<user_agent_label>) ‘ to the HTTP header when it is specified

Example invocation and results attached

[UserAgent.zip](https://github.com/lenovo/xclarity_client/files/1367734/UserAgent.zip)

### CODE REVIEW

**Comment by rodneyhbrown7** [comment]
Manasa it looks like some of the commits are signed by  root. Can you amend these commits to have a proper name/user?

**Comment by AndreyMenezes** [comment]
He need to squash this commits, basically one commit per line!

File: lib/xclarity_client/configuration.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 module XClarityClient
   class Configuration
 
-    attr_accessor :username, :password, :host, :auth_type, :port, :verify_ssl
+    attr_accessor :username, :password, :host, :auth_type, :port, :verify_ssl, :user_agent_label
+
```
Comment (AndreyMenezes) [inline lib/xclarity_client/configuration.rb]: Remove this blanc line plz!

File: spec/xclarity_client_config_profile_spec.rb [hunk 2]
```diff
@@ -114,7 +114,7 @@
       it 'unassigns the profile' do
         @client.unassign_config_profile(@idArray[0],"False", "False", "False")
         uri = "#{@host}/profiles/unassign/#{@idArray[0]}"
-        expect(a_request(:post, uri).with(:body => JSON.generate(force: 'False', powerDownITE: 'False', resetIMM: 'False'), :headers => {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'Authorization'=>'Basic Og==', 'Content-Type'=>'application/json', 'User-Agent'=>'Faraday v0.9.2'})).to have_been_made
+        expect(a_request(:post, uri).with(:body => JSON.generate(force: 'False', powerDownITE: 'False', resetIMM: 'False'))).to have_been_made
```
[the remark this answers is not in the document]
Comment (walteraa) [reply spec/xclarity_client_config_profile_spec.rb]: You should test the feature you are working in. If you are working in a Header feature, please don't remove the line that verifies the feature. 

You can use `FactoryGirl` to generate a Header example and use it in a test.

[the remark this answers is not in the document]
Comment (rodneyhbrown7) [reply spec/xclarity_client_config_profile_spec.rb]: @walteraa This comment is addressed now. Do you agree with this implementation?

[the remark this answers is not in the document]
Comment (walteraa) [reply spec/xclarity_client_config_profile_spec.rb]: @rodneyhbrown7 I agree. In another moment, when is need use this header in other contexts, I recommend adding the hash used in `: header` in a variable. For now is ok for me.

**Comment by walteraa** [comment]
LGTM after squashing the commits.

**Comment by walteraa** [comment]
@ManasaRaoK @rodneyhbrown7 @AndreyMenezes LGTM after squashing the commits and rebase.

**Pull request closed without merging**